pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -1403,7 +1403,9 @@ func (p *LogicalJoin) constructInnerIndexScanTask(
 		is.usedStatsInfo = usedStats.GetUsedInfo(is.physicalTableID)
 	}
 	finalStats := ds.tableStats.ScaleByExpectCnt(rowCount)
-	is.addPushedDownSelection(cop, ds, tmpPath, finalStats)
+	if err := is.addPushedDownSelection(cop, ds, tmpPath, finalStats); err != nil {
+		return nil
```

Comment:
If partition pruning fail, then `err != nil`. We'll not add the Selection, will this cause the wrong data returned?
And may other error happen and ignored here?

Reply:
Not sure, but https://github.com/pingcap/tidb/pull/52660/files/5343d4eb1dab5f57b431403234ad6f7f9df7cbee#diff-e82acde450b380d2b2347aeb83d34bffaaa10b19d9eb24ea7aadbef96d71c875R1246 also return a nil

Reply:
/cc @qw4990 @AilinKid

Reply:
Added a log here and the return nil means the innerPlan is empty. So I think no wrong data will returned.

Reply:
when the returned task is nil, it means the buildTaskForInner is failed. other alternatives will be chooses except this, it's ok